### PR TITLE
[hvtranslate]: refactor & performance improvements

### PR DESCRIPTION
**Pull request by SukkaW**

### CODE REVIEW

File: HVTranslate/HentaiVerse汉化.user.js [hunk 1]
```diff
@@ -2764,11 +2764,9 @@ var words = {
     // translatedList格式：key:已翻译元素, value: 该元素已被翻译属性和原文键值对（目前没考虑无法直接用key赋值的属性）
     //原文切换功能
     function restoreTranslate() {
-        translatedList.forEach(function (data, elem){
+        translatedList.forEach((data, elem) => {
             for (var item in data) {
-                var temp = elem[item];
-                elem[item] = data[item];
-                data[item] = temp;
+                [elem[item], data[item]] = [data[item], elem[item]];
```
Comment (SukkaW) [inline HVTranslate/HentaiVerse汉化.user.js:2769]: ES6 syntax for swap value.

File: HVTranslate/HentaiVerse汉化.user.js [hunk 2]
```diff
@@ -2839,28 +2837,33 @@ var words = {
 
     // function to decide whether a parent tag will have its text replaced or not
     function isTagOk(tag) {
-        return tagsWhitelist.indexOf(tag) === -1;
+        return !tagsWhitelist.includes(tag);
     }
 
     //翻译用到的字典变量
-    var regexps = {}; //存储转换过的字典，key值和word字典对应分组名相同，value格式见下方buildDict;
+    const regexps = new Map(); //存储转换过的字典，key值和word字典对应分组名相同，value格式见下方buildDict;
```
Comment (SukkaW) [inline HVTranslate/HentaiVerse汉化.user.js:2844]: `Map` performs better in scenarios involving frequent additions and removals of key-value pairs.

File: HVTranslate/HentaiVerse汉化.user.js [hunk 3]
```diff
@@ -2902,15 +2905,15 @@ var words = {
             });
         });
         //遍历分组字典
-        for (var selector in dictsMap) {
-            var elem = document.body.querySelector(selector);
+        for (const [selector, value] of Object.entries(dictsMap)) {
```
Comment (SukkaW) [inline HVTranslate/HentaiVerse汉化.user.js:2908]: https://eslint.org/docs/rules/guard-for-in

**Comment by indefined** [comment]
Thank you

**Pull request merged**